docs/src/userguide/faq.rst

```diff
@@ -0,0 +1,1018 @@
+
+But you can use Cython's `pure Python mode<pure-mode>`_ to provide type declarations for the compilation, including Python's PEP-484 syntax for type hints.
+
+For cases where no interaction with external C libraries is required, this is also the recommended way to type your code, since sticking to .py files with regular Python syntax keeps the whole range of debugging, linting, formatting, profiling etc. tools for Python code available for your software development needs, which usually cannot handle the syntax of .pyx files.
+
+----------
+
+Can Cython generate C code for classes?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: Yes, these classes become fully fledged Python classes.
+
+----------
+
+Can I call my Python code from C?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: Yes, easily. Follow the example in Demos/callback/ in the Cython source distribution.
+
+----------
+
+How do I interface numpy arrays using Cython?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: Follow the `example<numpy_tutorial>`_.
+
+----------
+
+How do I compile Cython with subpackages?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: It's highly recommended to arrange Cython modules in exactly the same Python package structure as the Python parts of the code base. As long as you don't keep your Cython code in unusual places, everything should just work.
+
+This is in part due to the fact that fully qualified names are resolved at compile time, and moving .so files around or adding __init__ files between the Cython compile and the Python runtime invocation means that cimports and imports may resolve differently. Failure to do this may result in errors like .pxd files not found or ``'module' object has no attribute '__pyx_capi__'``.
+
+----------
+
+How do I speed up the C compilation?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: Especially with large modules, the code that Cython generates can take the C compiler quite some time to optimise. This is usually ok for production builds, but during development, this can get in the way.
+
+It can substantially speed up the C compiler runs to disable the code optimisation, e.g. by setting the environment variable ``CFLAGS="-O0 -ggdb"`` on Linux or MacOS, which also enables full debugging symbols for better crash reports and debugger usage.  For MSVC on Windows, you can pass the option ``/Od`` to disable all optimisations.
+
+----------
+
+How do I reduce the size of the binary modules?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+**Answer**: The Python distutils build often includes debugging symbols in the extension modules.  The default for gcc is ``-g2``, for example. Disabling them (``CFLAGS=-g0`` for gcc), or setting them to the bare minimum that is required to produce stack traces on crashes (``CFLAGS=-g1`` for gcc), can visibly reduce the size of the binaries.
+
+----------
+
+How well is Unicode supported?
```

Comment:
I am pretty sure this chapter is no longer needed

Reply:
The first paragraph still looks relevant to me. The second paragraph should probably go since `Py_UNICODE` is deprecated